lib/prometheus/que_stats.rb

```diff
@@ -17,19 +17,48 @@ def worker_stats
       end
     end
 
-    def job_stats(filter = nil)
-      filter = "WHERE #{filter}" if filter
+    # ApplicationJob did not have to be here, but it's just harder to test otherwise because of ApplicationJob#delete_duplicates
+    JOB_CLASSES = %w[ApplicationJob ProcessEntryJob ProcessIntegrationEntryJob UpdateJob].inspect.tr('"', "'").freeze
+
+    def job_stats(*filters)
+      filter = "WHERE #{filters.join(' AND ')}" if filters.presence
       sql = <<~SQL
-        SELECT args->0->>'job_class' AS job_class, COUNT(*) as count
-        FROM que_jobs #{filter}
-        GROUP BY args->0->>'job_class'
+        WITH
+          jobs AS (SELECT unnest(array#{JOB_CLASSES}) AS job_class),
+          stats AS (
+            SELECT args->0->>'job_class' AS job_class, COUNT(*) as count
+            FROM que_jobs #{filter}
+            GROUP BY args->0->>'job_class'
+          )
+        SELECT jobs.job_class, COALESCE(stats.count, 0) AS count
+        FROM jobs LEFT JOIN stats ON jobs.job_class = stats.job_class
       SQL
 
       execute do |connection|
         connection.select_all(sql)
       end
     end
 
+    def job_stats_ready
+      job_stats('error_count = 0', 'expired_at IS NULL', 'finished_at IS NULL', 'run_at <= now()')
```

Comment:
Could be:

```ruby
# You can keep last condition instead of using Time.zone.now if you want
conditions = [ {error_count: 0, expired_at: nil, finished_at: nil}, ['run_at <= ?', Time.zone.now]]

job_stats(conditions)

```

Reply:
Following https://github.com/3scale/zync/pull/334/files#r429103152.